### PR TITLE
Line class added

### LINKED ISSUE #268: [✨ Feature request]: EPIC: Rectangular Reinforced Cross Section
### Description

Implement a Rectangular Reinforced Cross Section. In the future to be used with a separate Analysis class.

### Acceptance Criteria
- [x] Concrete Material;
- [x]  Rebar Material;
- [ ]  Rebar object (material, location etc.);
- [ ]  (Reinforced) CrossSection Abstract Baseclass or Protocol (area, perimeter, list of nodes?, section moduli, rebars, cover_definitions);
- [ ]  ReinforcedRectangularCrossSection definition; 

### Dependencies

_No response_

### Confirmation

- [ ]  I have checked that this feature does not already exist in the issue tracker.
- [ ]  I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by egarciamendez**
## Description

An objecte representation of a line.

Fixes #268 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: blueprints/geometry/line.py [hunk 1]
```diff
@@ -1 +1,229 @@
...
+            msg = "n must be equal or greater than 2"
+            raise ValueError(msg)
+
+        evenly_spaced_points = self.get_evenly_spaced_points(n + 1)
+        return [Line(start_point=point_1, end_point=point_2) for point_1, point_2 in zip(evenly_spaced_points[:-1], evenly_spaced_points[1:])]
+
+    def __eq__(self, other: object) -> bool:
+        """Return True if the lines are equal."""
+        if not isinstance(other, Line):
+            raise NotImplementedError
```
Comment (bro-wi) [inline blueprints/geometry/line.py:228]:
```suggestion
            raise NotImplementedError("Line can only be compared to other Line object")
```

File: blueprints/geometry/line.py [hunk 2]
```diff
@@ -1 +1,229 @@
...
+        return np.array_equal(self._start, other._start) and np.array_equal(self._end, other._end)
```
Comment (bro-wi) [inline blueprints/geometry/line.py:229]: shouldn't we compare using np.allclose() ? comparing floats can be nasty

**Review by bro-wi — APPROVED** [review]
Great improvement on the previous version! Only some minor remarks.

File: blueprints/geometry/line.py [hunk 3]
```diff
@@ -1 +1,229 @@
...
+
+    id: int = 0
+
+    def __init__(self, start_point: Point, end_point: Point) -> None:
+        self.start_point = start_point
+        self.end_point = end_point
+        self._start = np.array(start_point.coords)
+        self._end = np.array(end_point.coords)
+        self._validate_points()
+        Line.id += 1
```
[the remark this answers is not in the document]
Comment (egarciamendez) [reply blueprints/geometry/line.py:40]: You're right. Good call.

File: blueprints/geometry/line.py (hunk 2 above)
Comment (egarciamendez) [reply blueprints/geometry/line.py:229]: 👍

File: blueprints/geometry/line.py [hunk 4]
```diff
@@ -1 +1,229 @@
...
+        distance : float
+            Distance to extend or shorten the line. Positive number extends the line, negative number shortens the line.
+        direction: Literal["start", "end"]
+            Given direction where the line needs to be extended. Default towards the end of the line.
+
+        Returns
+        -------
+        Point
+            Extended/shortened point of the line.
+        """
```
[the remark this answers is not in the document]
Comment (egarciamendez) [reply blueprints/geometry/line.py:170]: In hindsight it is indeed not that intuitive👍

**Comment by egarciamendez** [comment]
@bro-wi Thanks for your review :-) I have made the changes accordingly

**Pull request merged**